models/src/main/scala/coop/rchain/models/rholangn/Basic.scala

```diff
@@ -136,11 +152,21 @@ object MatchCaseN {
 final class NewN(
     val bindCount: Int,
     val p: ParN,
-    val uri: Seq[String],
-    val injections: Map[String, ParN]
+    val uri: Seq[GStringN],
+    val injections: Map[GStringN, ParN]
 ) extends BasicN {
-  def sortedUri: Seq[String]                = parmanager.Manager.sortUris(uri)
-  def sortedInjections: Seq[(String, ParN)] = parmanager.Manager.sortInjections(injections)
+  // Sorted by objects hash which is memoized as Rho type
+  val urisSorted: Eval[Seq[GStringN]] =
+    this.uri.traverse(uri => uri.rhoHash.map((uri, _))).map(_.sortBy(_._2).map(_._1)).memoize
+
+  // Sorted by objects hash which is memoized as Rho type
+  val injectionsSorted: Eval[Seq[(GStringN, ParN)]] =
+    this.injections.toSeq
+      .traverse(inj => inj.bimap(_.rhoHash, _.rhoHash).mapN(_ ++ _).map((inj, _)))
+      .map(_.sortBy(_._2).map(_._1))
```

Comment:
I think this would be easier to read 
`_.sortBy(_._2).map(_._1)` 
=> 
`_.sortBy{ case (_, hash) => hash }.map{ case (inj, _) => inj }`